src/RMGVertexConfinement.cc

```diff
@@ -121,8 +121,129 @@ bool RMGVertexConfinement::SampleableObject::IsInside(const G4ThreeVector& verte
 
   return false;
 }
+
+
+// get the intersections between the solid and an arbitrary line, defined by start and dir
+std::vector<G4ThreeVector> RMGVertexConfinement::SampleableObject::GetIntersections(G4ThreeVector start,
+    G4ThreeVector dir) const {
+
+
+  // check if the physical volume exists
+  if ((not this->physical_volume) or (not this->physical_volume->GetLogicalVolume()->GetSolid()))
+    RMGLog::OutDev(RMGLog::fatal, "Cannot find number of intersections for a SampleableObject ",
+        "where the physical volume is not set this probably means you are trying to generically ",
+        "sample a geometrical volume which instead should be natively sampled");
+
+  auto solid = this->physical_volume->GetLogicalVolume()->GetSolid();
+
+
+  G4double dist = 0;
+  std::vector<G4ThreeVector> intersections = {};
+  G4ThreeVector int_point;
+
+  int_point = start;
+  int counter = 0;
+
+  dist = 0;
+
+  while (dist < kInfinity) {
+
```

Comment:
all these double newlines are not needed here
```suggestion

// get the intersections between the solid and an arbitrary line, defined by start and dir
std::vector<G4ThreeVector> RMGVertexConfinement::SampleableObject::GetIntersections(G4ThreeVector start,
    G4ThreeVector dir) const {

  // check if the physical volume exists
  if ((not this->physical_volume) or (not this->physical_volume->GetLogicalVolume()->GetSolid()))
    RMGLog::OutDev(RMGLog::fatal, "Cannot find number of intersections for a SampleableObject ",
        "where the physical volume is not set this probably means you are trying to generically ",
        "sample a geometrical volume which instead should be natively sampled");

  auto solid = this->physical_volume->GetLogicalVolume()->GetSolid();

  G4double dist = 0;
  std::vector<G4ThreeVector> intersections = {};
  G4ThreeVector int_point;

  int_point = start;
  int counter = 0;

  dist = 0;

  while (dist < kInfinity) {
```